### DISCUSSION TITLE
next lint in lint-staged always lints whole project

### DISCUSSION THREAD

**Discussion by nichtsam** [vercel/next.js#38632] [Help]
https://nextjs.org/docs/basic-features/eslint#lint-staged

I use the way provided in the document.
However I found that it still lint the whole project, the `--file` flag only attach additional files to it.

I use lint-staged to only lint the files that staged, isn't this pretty against it? 
or am I doing something wrong here?

**Comment by sianlasaga** [comment]
Same problem here. The `--file` flag that should check the target file only doesn't work.

**Reply by nichtsam** [reply]
If I remember correctly, it does check the target files.
However, it checks them in addition to the source code, which is not the desired behavior.
